Projects/Application/Sources/Common/Network/Http/Http.swift

```diff
@@ -0,0 +1,51 @@
+//
+// Created by Maxim Berezhnoy on 2019-02-12.
+// Licensed under the MIT license
+//
+//
+
+import Foundation
+
+// MARK: - RequestAdapter
+protocol RequestAdapter {
+    func adapt(urlRequest: URLRequest) -> URLRequest
+}
+
+// MARK: - RequestRetrier
+protocol RequestRetrier {
+    func should(retry request: Http.Request, with response: Http.Response, retryAfter: (TimeInterval) -> Void)
+}
+
+// MARK: - Http core
+final class Http {
```

Comment:
I like the idea to use `HTTP` as a namespace, but not like a name for HTTP client. Maybe the name can be changed to something similar to `Provider` like in Moya.

Reply:
I was going for something simple and concise like `Http.request("www.google.com")`.
However, I agree, naming should better reflect the intent.

Reply:
Renamed to `Client` (`HTTP.Client(configuration).request` seems to be clear enough)